src/kimmdy/topology/topology.py

```diff
@@ -86,7 +89,25 @@ def __init__(self, header: tuple[str, str], atomics: dict) -> None:
         self._parse_settles()
         self._parse_exclusions()
         self._initialize_graph()
-        self.test_for_radicals()
+
+        # must be after self._parse_atoms
+        self.radicals: dict[str, Atom] = {}
+        if radicals is not None:
+            if self.name == "Reactive":
+                logger.debug(
+                    f"Using 'radicals' section from config file with entries: '{radicals}'."
+                )
+            for radical in radicals.split(sep=" "):
+                atom = self.atoms.get(radical)
+                if atom:
+                    self.radicals[radical] = atom
+                else:
+                    logger.debug(
+                        f"Atom nr {radical} from 'radicals' section in config file not in topology. Ignoring this entry."
+                    )
+        else:
+            self.find_radicals()
+        self.set_atom_is_radical()
```

Comment:
Ah, I see now how these functions have to be used independently of each other, so it's just about finding better names, not merging them.

Reply:
the `set_atom_is_radical` method was not necessary in the end. Refactored into the `find_radicals` method and the `Moleculetype.__init__`